components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -82,11 +87,23 @@ public void addClaimDialect(ClaimDialect claimDialect, String tenantDomain) thro
         if (claimDialect == null || StringUtils.isBlank(claimDialect.getClaimDialectURI())) {
             throw new ClaimMetadataClientException(ERROR_CODE_EMPTY_CLAIM_DIALECT);
         }
+        if (StringUtils.isBlank(tenantDomain)) {
+            throw new ClaimMetadataClientException(ERROR_CODE_EMPTY_TENANT_DOMAIN);
+        }
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (tenantId == MultitenantConstants.INVALID_TENANT_ID) {
+            throw new ClaimMetadataClientException(ERROR_CODE_INVALID_TENANT_DOMAIN.getCode(),
+                    String.format(ERROR_CODE_INVALID_TENANT_DOMAIN.getMessage(), tenantDomain));
+        }
 
-        // TODO : validate claim dialect already exists?
+        List<ClaimDialect> claimDialects = this.claimDialectDAO.getClaimDialects(tenantId);
+        List<String> claimDialectUris = claimDialects.stream().map(ClaimDialect::getClaimDialectURI).
```

Comment:
since we are using contains() below, better to use a set here which will be efficient

Reply:
Fixed: e892da0880f754ef15e877f078706ba84d41dd8e